src/decorator/date/MaxDate.ts

```diff
@@ -17,7 +18,7 @@ export function MaxDate(date: Date | (() => Date), validationOptions?: Validatio
   return ValidateBy(
     {
       name: MAX_DATE,
-      constraints: [date],
+      constraints: [isDate(date) ? date : date()],
```

Comment:
I have found isDate typechecker and used it to see whether passed prop is Date or function which returns Date. I think it is safe to use here, right?